### PR TITLE
Use zip instead of tar/gzip for the default export format

### PR DESCRIPTION
**Pull request by dehort**
RHCLOUD-33208

Needs more testing...I ripped out the function and tested it manually/directly (I hacked it together to run outside of the service) because podman is giving me trouble.

Not sure about compression...

We really need to break this function up.  Right now it has too many responsibilites.  It downloads the files from s3, builds the zip file, then uploads the zip file to s3.  Splitting it up would make it easier to understand and easier to test.

Some info on compression:  https://tanaikech.github.io/2018/10/23/zip-compression-of-downloaded-file-using-golang/

## What?
Explain what the change is linking any relevant JIRAs or Issues.

## Why?
Consider what business or engineering goal does this PR achieves.

## How?
Describe how the change is implemented. Any noteable new libaries, APIs, or features.

## Testing
Did you add any tests for the change?

## Anything Else?
Any other notes about the PR that would be useful for the reviewer. 

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [X] General Coding Practices

### CODE REVIEW

File: s3/compress.go [hunk 1]
```diff
@@ -101,16 +99,17 @@ func (c *Compressor) zipExport(ctx context.Context, prefix, filename, s3key stri
...
 
-		header, err := tar.FileInfoHeader(fi, basename)
+		header, err := zip.FileInfoHeader(fi)
 		if err != nil {
 			return fmt.Errorf("failed to create file header: %w", err)
 		}
 		header.Name = basename
 
-		if err = tarWriter.WriteHeader(header); err != nil {
+		var zippedFile io.Writer
```
Comment (skarekrow) [inline s3/compress.go:108]: This should be defined at the top for consistent style.

File: s3/compress.go [hunk 2]
```diff
@@ -101,16 +99,17 @@ func (c *Compressor) zipExport(ctx context.Context, prefix, filename, s3key stri
...
+		if zippedFile, err = zipWriter.CreateHeader(header); err != nil {
```
Comment (skarekrow) [inline s3/compress.go:109]: Do we even need the above? 
```suggestion
		if zippedFile, err := zipWriter.CreateHeader(header); err != nil {
```
Would this not work?

File: s3/compress.go (hunk 1 above)
Comment (dehort) [reply s3/compress.go:108]: I want to limit the scope of that variable so I am creating/instantiating it as far down as possible (to just where it is used).

File: s3/compress.go (hunk 2 above)
Comment (dehort) [reply s3/compress.go:109]: No, that approach will not work as it is currently coded.  With the approach you are suggesting, the `zippedFile` variable will be scoped to that if statement, but the `zippedFile` variable is needed outside of that if statement.

Another approach, would be to split the if statement like this:
`
                zippedFile, err := zipWriter.CreateHeader(header); 
		if err != nil {
`

**Comment by dehort** [comment]
Looking closer at this code...I think this might be leaking file handles

**Pull request merged**